client/src/pages/Login.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useState } from "react";
+import { TextInputField } from "./../components/Components";
+import { Container, Grid, Typography, Button } from "@material-ui/core";
+import { IdentityToken, emptyIdentityToken } from "./../interfaces";
+import Logo from "./../img/Logo.png";
+import "./../css/Login.css";
+
+// Applease typescript
+declare const window: any;
+// Grab handle to our microapps js library
+const microapps = window.microapps;
+
+function Login() {
+  const is_web = window.location == window.parent.location;
```

Comment:
(Please file a new issue and leave TODO )
if this check is being done in multiple places, then you should extract it out to a common function!
Also, it might be better to check if window is inside iFrame instead of comparing with the urls e.g.
```
 window.self !== window.top;
```

Reply:
resolved in #22